### ISSUE TITLE
Changed format of result file

### ISSUE THREAD

**Issue by capiman** [msoos/cryptominisat#296]
Earlier it looked like

SAT
Solution 1
Solution 2
Solution 3

Now

SAT
Solution 1
SAT
Solution 2
SAT
Solution 3
UNSAT

From my point of view, the new format look nicer than the old one.
Bug or feature? I think it is no problem for me, perhaps other programs
will not understand the result file out-of-the-box any more
and report errors/wrong results parsing the result file.

**Comment by msoos** [comment]
Feature :) I think it looks better! I hope it's OK, closing.

**Comment by egri-nagy** [comment]
It certainly looks nice, but I need to store billions of solutions  and process them. For this use case one line per solution would be the best.
Would it be possible to have a switch for pretty printing (with default on)? so I could have the raw output as well.
I can of course store and parse the output as it is now.

**Comment by msoos** [comment]
I think for now I don't really intendt to change the format. Maybe you could introduce a new format that is binary? That'd be a LOT smaller and a lot faster to parse. I suggest you have 4bits/variable, as l_Undef needs to be represented as well as SAT and UNSAT. You'll have a reduction in the order of 10-100x. That'd be bit-packed. Another one is to byte-pack, 4bytes/variable, and a variable is only in if it's not l_Undef. Special (e.g. 0xfffff... is SAT 0xffff-1 is UNSAT). You can implement it here:

```
if (ret == l_True && (printResult || toFile)) {
        if (toFile) {
            for (uint32_t var = 0; var < solver->nVars(); var++) {
                if (solver->get_model()[var] != l_Undef) {
                    *os << ((solver->get_model()[var] == l_True)? "" : "-") << var+1 << " ";
                }
            }
            *os << "0" << endl;
        } else {
            const uint32_t num_undef = print_model(os, solver);
            if (num_undef && !toFile && conf.verbosity) {
               cout << "c NOTE: " << num_undef << " varables are NOT set" << endl;
            }
        }
    }
```

**Comment by egri-nagy** [comment]
Thanks! I see it now that my requirements are rather special, and have to use cryptominisat as a library anyway. 
Basically I only need the set of variables that are set to true. And we know that these sets tend to have small size, so by ignoring false assignments we can already save a lot.
In another project we used a bit-packed output, but still in a text file. We packed 6 bits into a character, to have maximum portability. But as mentioned before, we will probably be fine by just writing out true assignments.
